### ISSUE TITLE
Proposal/Drivers: Support for additional security permissions

### ISSUE THREAD

**Issue by andrestc** [docker/machine#3642]
It would be useful if the core drivers provided a way for the user to add additional security permissions(opened ports on the host). 

My use case: I'm currently using docker machine drivers to automate the installation of multiple pre-determined services with known exposed ports that should be accessible from the outside. In order to have my workflow fully automated I need to create the security group manually (before starting the provision process) and then feeding it to the ec2 driver with `amazonec2-security-group`. This can be error prone and boring.

My proposal is to add a new, optional parameter: `amazonec2-security-rule` where one could configure additional rules to be appended to the default security group.

I believe we could support this kind of feature on every core driver that has some kind of security permission and would be happy to submit a PR to each of them.

**Comment by andrestc** [comment]
Hi @nathanleclaire, do you think a PR adding this kind of support would be merged into docker-machine? We are in need of this feature and can solve it either by contributing it docker-machine or handling this feature for each driver separately in our tool.

**Comment by nathanleclaire** [comment]
`--amazonec2-security-group` doesn't work for your needs ?

```
   --amazonec2-security-group [--amazonec2-security-group option --amazonec2-security-group option]	AWS VPC security group [$AWS_SECURITY_GROUP]
```

**Comment by andrestc** [comment]
Actually, no. I'm building a tool that uses docker-machine and its drivers to bootstrap docker hosts and start some services on them. I can't rely on having my end user create the security group needed before running my tool. So I need to provide a way to automatically add this security permissions.

**Comment by andrestc** [comment]
So I think I have two options: (1) implement this logic after the machine is provisioned (using the aws client(and the azure client etc etc) directly in my tool) or (2) adding this kind of feature to every driver it fits.

**Comment by andrestc** [comment]
Any thoughts on this @nathanleclaire?

**Comment by ahmetalpbalkan** [comment]
@andrestc for azure we create the Network Security Group for each machine and open Docker/SSH ports by default. We already get `--azure-open-port` user input to open extra ports (I believe this part is missing in other drivers). Would having such a generic create flag and having drivers to implement it solve your problem?

**Comment by andrestc** [comment]
@ahmetalpbalkan, yes, that would definitely solve my problem! I could get these PR's going if everyone is cool with it.

**Comment by ahmetalpbalkan** [comment]
@andrestc the only problem I see is that won't be applicable to all drivers:  Such as local virtualization software and (if I'm not mistaken) some clouds like DigitalOcean do not let you open ports selectively (instead they open them all).

So I think it would be good for now to add `--[drivername]-open-port` (array) flag to Google, AWS etc. Azure already has it.

**Comment by andrestc** [comment]
Yup, thats what i meant.

**Comment by nathanleclaire** [comment]
I'd be amenable to adding it to `$DRIVER` if you'd like to submit the PR

**Comment by andrestc** [comment]
@nathanleclaire, will do it!

**Comment by CharmingSteve** [comment]
I AM trying to add ports to security automatically to allow mongodb ports with replication
